tests/unit/cluster/shardid-propagation.tcl

```diff
@@ -53,3 +53,84 @@ tags {tls:skip external:skip cluster singledb} {
         }
     }
 }
+
+# R0 is Node A (cluster-allow-replica-migration no), R3 is Node B, R4 is Node C
+# 1. Node A goes down, and Node B takes over primaryship.
+# 2. Node A continues to be down while another Node C is added as a replica of B.
+# 3. Node B goes down, and Node C takes over primaryship.
+# 4. Node A and Node B come back up and start learning about the topology.
+# 5. Node A comes up thinking it was the primary (but has an older config epoch compared to C).
+# 6. Node A learns about Node C via gossip and assigns it a random `shard_id`.
+# 7. Node A receives a direct ping from Node C.
+#    a. Node C advertises the same set of slots that Node A was earlier owning.
+#    b. Since Node A assigns a random shard ID to Node C, Node A thinks that it is still a primary and
+#       it lost all its slots to Node C, which is in another shard.
+# 8. Node A then updates the actual `shard_id` of Node C while processing `shard_id` in ping extensions.
+# 9. Node A and Node C end up being primaries in the same shard while Node C continues to own slots.
+start_cluster 3 2 {tags {external:skip cluster}} {
```

Comment:
We can put some config we want for all nodes here.

```suggestion
start_cluster 3 2 {tags {external:skip cluster} overrides {allow-replica-migration no}} {
```

Reply:
yes, we can, here i am only explicitly setting for R A